src/common/snippets/src/lowered/pass/insert_reg_spills.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/insert_reg_spills.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/op/reg_spill.hpp"
+#include "snippets/op/brgemm.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/utils/utils.hpp"
+
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+bool InsertRegSpills::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InsertRegSpills")
+
+    auto needs_reg_spill = [](const ExpressionPtr& expr) {
+        return ov::is_type<snippets::op::Brgemm>(expr->get_node());
+    };
+    bool modified = false;
+    for (auto it = linear_ir.begin(); it != linear_ir.end(); it++) {
+        const auto& expr = *it;
+        if (!needs_reg_spill(expr))
+            continue;
+        auto start_it = std::prev(it);
+        auto stop_it = std::next(it);
+        while (ov::is_type<snippets::op::LoopBegin>(start_it->get()->get_node()) &&
+               ov::is_type<snippets::op::LoopEnd>(stop_it->get()->get_node())) {
+            start_it--;
+            stop_it++;
+        }
+        // Note: we need to insert immediately before LoopBegin => increment start_it
+        start_it++;
+        const auto& loop_begin_live = start_it->get()->get_live_regs();
+        std::set<Reg> brgemm_used;
+        const auto& brgemm_reg_info = expr->get_reg_info();
+        brgemm_used.insert(brgemm_reg_info.first.begin(), brgemm_reg_info.first.end());
+        brgemm_used.insert(brgemm_reg_info.second.begin(), brgemm_reg_info.second.end());
```

Comment:
I think we should avoid naming `brgemm` in common pass because not only `brgemm` can need reg spills (for example, `BrgemmCopyB` in CPU plugin). What's about to just call `expr` or `target_expr` instead of `brgemm`?